docs/getting-started/build-your-own-development-automation.md

```diff
@@ -217,6 +225,175 @@ Click on `Close` (you could also close the issue in GitHub if you prefer) and se
+yourorg:dotnetgettingstarted:0.1.0 ← local
+```
+
+Congratulations, you now have a new generator. It's usually a good idea at this point to give things a test run and so run the generator in the `rug shell` using the `generate` command, for example:
+
+```
+rug → generate NewDotNetASPCoreProject newdotnetservice -C ../
+```
+
+The `-C` changes the directory to above where you're running the shell for the output of the generator. Go now and check out the output and make sure it is still a working project. You should now be able to change directory to where you generated your new project and be able to test and run it as usual.
+
+That's the new generator created, now it's time to publish it so you can list and execute it using `@atomist` from your team's slack.
+
+### Publishing Your New Project Generator
+
+Similar to how you published your handlers project previously, this time you're going to publish your new generator rug archive. Execute the `rug publish` command from inside your generator's project directory:
+
+```
+> rug publish
+Resolving dependencies for yourorg:dotnetgettingstarted:0.1.0 ← local completed
+Invoking TypeScript Compiler on ts script sources
+  Created .atomist/editors/NewDotNetASPCoreProject.js.map
+  Created .atomist/editors/NewDotNetASPCoreProject.js
+Processing script sources completed
+Loading yourorg:dotnetgettingstarted:0.1.0 ← local into runtime completed
+  Created META-INF/maven/yourorg/dotnetgettingstarted/pom.xml
+  Created .atomist/manifest.yml
+  Created .atomist/metadata.json
+Generating archive metadata completed
+  Uploading yourorg/dotnetgettingstarted/0.1.0/dotnetgettingstarted-0.1.0.zip → t3v0s7ss2 (155 kb) succeeded
+  Uploading yourorg/dotnetgettingstarted/0.1.0/dotnetgettingstarted-0.1.0.pom → t3v0s7ss2 (659 bytes) succeeded
+  Uploading yourorg/dotnetgettingstarted/0.1.0/dotnetgettingstarted-0.1.0-metadata.json → t3v0s7ss2 (897 bytes) succeeded
+  Downloading yourorg/dotnetgettingstarted/maven-metadata.xml ← t3v0s7ss2 (394 bytes) succeeded
+  Uploading yourorg/dotnetgettingstarted/maven-metadata.xml → t3v0s7ss2 (346 bytes) succeeded
+Publishing archive into remote repository completed
+
+→ Archive
+  ~/code/src/github.com/yourorg/dotnetgettingstarted/.atomist/target/dotnetgettingstarted-0.1.0.zip (155 kb in 128 files)
+
+→ URL
+  https://atomist.jfrog.io/atomist/T3V0S7SS2/yourorg/dotnetgettingstarted/0.1.0/dotnetgettingstarted-0.1.0.zip
+```
+
+With your new generator's rug archive published it's now time to switch over to your team's slack again. Before you can see your new generator you will need to ask `@atomist` to refresh the rugs it is aware of by executing `@atomist refresh rugs`:
+
+<div class="ss-container">
+  <img src="../images/refreshing-rugs.png" alt="Refreshing the Rugs that Atomist is aware of" class="ss-small">
+</div>
+
+Now you can list the generators using `@atomist generators` and you should see your new generator in the list for your team:
```

Comment:
Won't they have to provide a search term to make sure it will be in those shown?